front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -66,6 +67,69 @@ const usePathfinding = ({
   const [postPathProperties] =
     osrdEditoastApi.endpoints.postInfraByInfraIdPathProperties.useLazyQuery();
 
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+
+  const fetchOperationalPoints = useCallback(
+    async (steps: PathStep[]) => {
+      if (!infraId || !steps.length) return;
+
+      const searchPayload = buildOpSearchQuery(
+        infraId,
+        steps.filter((step) => step !== null)
+      );
+
+      if (!searchPayload) return;
+
+      try {
+        const results = (await postSearch({
+          searchPayload,
+          pageSize: 101,
+        }).unwrap()) as SearchResultItemOperationalPoint[];
+
+        if (!results.length) return;
+
+        const updatedSteps = steps.map((step) => {
+          if (!step) return step;
+
+          let matchedOp: SearchResultItemOperationalPoint | undefined;
+
+          const stepUICString = 'uic' in step && step.uic ? String(step.uic) : null;
+
+          if (stepUICString) {
+            matchedOp = results.find(
+              (op) =>
+                `87${op.ci}` === stepUICString &&
```

Comment:
I'd prefer to avoid hardcoding French-specific assumptions if we can. We should be able to use `op.uic` here instead.